client/snap_op.go

```diff
@@ -55,30 +56,38 @@ func writeFieldBool(mw *multipart.Writer, key string, val bool) error {
 	return mw.WriteField(key, "true")
 }
 
+type field struct {
+	field string
+	value bool
+}
+
+func writeFields(mw *multipart.Writer, fields *[]field) error {
```

Comment:
pointer to slices are very unusual, unless you need to write over fields itself, I would expect []field here